book/src/dev/rfcs/0005-treestate.md

```diff
@@ -0,0 +1,194 @@
+metadata (for Sapling) or previously finalized state (for Sprout). This document
+describes how we compute and manage that data, assuming a finalized state
+service as described in the [State Updates RFC](https://zebra.zfnd.org/dev/rfcs/0005-state-updates.md).
+
+
+# Motivation
+[motivation]: #motivation
+
+Block validation requires checking that the treestate of the block (consisting
+of the note commitment tree and nullifier set) is consistent with the metadata
+we have in the block header (the root of the note commitment tree) or previously
+finalized state (for Sprout).
+
+
+# Definitions
+[definitions]: #definitions
+
+Many terms used here are defined in the [Zcash Protocol Specification](https://zips.z.cash/protocol/protocol.pdf)
+
+**notes**: Represents a value bound to a shielded payment address (public key)
+which is spendable by the recipient who holds the spending key corresponding to
+a given shielded payment address.
+
+**nullifiers**: Revealed by Spend descriptions when its associated note is spent.
+
+**nullifier set**: The set of unique nullifiers revealed by any transactions
+within a block. Nullifiers are enforced to be unique within a valid block chain
+by commiting to previous treestates in Spend descriptions, in order to prevent
+double-spends.
+
+**note commitments**: Pedersen commitment to the values consisting a note. One
+should not be able to construct a note from its commitment.
+
+**note commitment tree**: An incremental Merkle tree of fixed depth used to
+store note commitments that JoinSplit transfers or Spend transfers produce. It
+is used to express the existence of value and the capability to spend it. It is
+not the job of this tree to protect against double-spending, as it is
+append-only: that's what the nullifier set is for.
+
+**note position**: The index of a note commitment at the leafmost layer,
+counting leftmost to rightmost. The [position in the tree is determined by the
+order of transactions in the block](https://zips.z.cash/protocol/canopy.pdf#transactions).
+
+**root**: The layer 0 node of a merkle tree.
+
+**anchor**: A Merkle tree root of a note commitment tree. It uniquely identifies
+a note commitment tree state given the assumed security properties of the Merkle
+tree’s hash function.  Since the nullifier set is always updated together with
+the note commitment tree, this also identifies a particular state of the
+associated nullier set.
+
+**spend descriptions**: A shielded Sapling transfer that spends a note. Includes
+an anchor of some previous block's note commitment tree.
+
+**output descriptions**: A shielded Sapling transfer that creates a
+note. Includes the u-coordinate of the note commitment itself.
+
+**joinsplit**: A shielded transfer that can spend Sprout notes and transparent
+value, and create new Sprout notes and transparent value, in one Groth16 proof
+statement.
+
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
```

Comment:
It might be helpful to split the guide into Sprout and Sapling sections, and a common section.

Reply:
and also an Orchard section now?

Reply:
👍 added as a todo in the draft
https://github.com/ZcashFoundation/zebra/pull/983/commits/5e39bc328593e5f20338010ed607a2afdafa4820